### ISSUE TITLE
Retry on EPIPE

### ISSUE THREAD

**Issue by arunoda** [Azure/azure-cosmos-js#399]
We see an error coming from the client with EPIPE as the error code. Here's a such an error:

```
{ Error: write EPIPE
    at WriteWrap.afterWrite [as oncomplete] (net.js:779:14)
  [stack]:
   'Error: write EPIPE\n    at WriteWrap.afterWrite [as oncomplete] (net.js:779:14)',
  [message]: 'write EPIPE',
  errno: 'EPIPE',
  code: 'EPIPE',
  syscall: 'write',
  headers:
   { 'x-ms-request-charge': 29.980000000000004,
     'x-ms-documentdb-query-metrics': {} } }
```

If we retry this, we will get the expected result. So, I assume this is something the client should handle. 

We could easily reproduce this when we load testing our APIs. So, I assume this is coming when there are many HTTP requests talking to the server.

Let me know, if you know any additional details.

**Comment by southpolesteve** [comment]
Fixed in 3.1.1

**Comment by arunoda** [comment]
Thanks @southpolesteve
